examples/textinput.py

```diff
@@ -6,169 +6,235 @@
 
 # Version check
 if pg.get_sdl_version() < (2, 0, 0):
     raise Exception("This example requires pygame 2.")
 
-###CONSTS
-# Set to true or add 'showevent' in argv to see IME and KEYDOWN events
-PRINT_EVENT = False
-# frames per second, the general speed of the program
-FPS = 50
-# size of window
-WINDOWWIDTH, WINDOWHEIGHT = 640, 480
-BGCOLOR = (0, 0, 0)
-
-# position of chatlist and chatbox
-CHATLIST_POS = pg.Rect(0, 20, WINDOWWIDTH, 400)
-CHATBOX_POS = pg.Rect(0, 440, WINDOWWIDTH, 40)
-CHATLIST_MAXSIZE = 20
-
-TEXTCOLOR = (0, 255, 0)
-
-# Add fontname for each language, otherwise some text can't be correctly displayed.
-FONTNAMES = [
-    "notosanscjktcregular",
-    "notosansmonocjktcregular",
-    "notosansregular,",
-    "microsoftjhengheimicrosoftjhengheiuilight",
-    "microsoftyaheimicrosoftyaheiuilight",
-    "msgothicmsuigothicmspgothic",
-    "msmincho",
-    "Arial",
-]
-
-# Initalize
-pg.init()
-Screen = pg.display.set_mode((WINDOWWIDTH, WINDOWHEIGHT))
-pg.display.set_caption("TextInput example")
-FPSClock = pg.time.Clock()
-
-# Freetype
-# "The font name can be a comma separated list of font names to search for."
-FONTNAMES = ",".join(str(x) for x in FONTNAMES)
-Font = freetype.SysFont(FONTNAMES, 24)
-FontSmall = freetype.SysFont(FONTNAMES, 16)
-print("Using font: " + Font.name)
-
-# Main loop process
-def main():
-    global BGCOLOR, PRINT_EVENT, CHATBOX_POS, CHATLIST_POS, CHATLIST_MAXSIZE
-    global FPSClock, Font, Screen
 
+class TextInput:
     """
-    https://wiki.libsdl.org/SDL_HINT_IME_INTERNAL_EDITING
-    https://wiki.libsdl.org/Tutorials/TextInput
-    Candidate list not showing due to SDL2 problem ;w;
+    A simple TextInput class that allows you to receive inputs in pygame.
     """
-    pg.key.start_text_input()
-    input_rect = pg.Rect(80, 80, 320, 40)
-    pg.key.set_text_input_rect(input_rect)
-
-    _IMEEditing = False
-    _IMEText = ""
-    _IMETextPos = 0
-    _IMEEditingText = ""
-    _IMEEditingPos = 0
-    ChatList = []
-
-    while True:
-        for event in pg.event.get():
-            if event.type == pg.QUIT:
-                pg.quit()
-                return
-
-            elif event.type == pg.KEYDOWN:
-                if PRINT_EVENT:
+
+    # Add font name for each language,
+    # otherwise some text can't be correctly displayed.
+    FONT_NAMES = [
+        "notosanscjktcregular",
+        "notosansmonocjktcregular",
+        "notosansregular,",
+        "microsoftjhengheimicrosoftjhengheiuilight",
+        "microsoftyaheimicrosoftyaheiuilight",
+        "msgothicmsuigothicmspgothic",
+        "msmincho",
+        "Arial",
+    ]
+
+    def __init__(
+        self, promt: str,
+        pos, screen_dimensions,
+        print_event: bool,
+        text_color="white"
+    ) -> None:
+        self.promt = promt
+        self.print_event = print_event
+        # position of chatlist and chatbox
+        self.CHAT_LIST_POS = pg.Rect(pos, (screen_dimensions[0], 400))
```

Comment:
Use a slightly different position for the CHAT_LIST_POS (+50 in y direction would do the trick I guess)

Reply:
Got it, thanks for the review!